frappe/model/meta.py

```diff
@@ -532,7 +532,7 @@ def get_high_permlevel_fields(self):
 
 		return self.high_permlevel_fields
 
-	def get_permitted_fieldnames(self, parenttype=None, *, user=None):
+	def get_permitted_fieldnames(self, parenttype=None, *, user=None, permission_type="read"):
```

Comment:
```suggestion
	def get_permitted_fieldnames(self, parenttype=None, *, user=None, permission_type=None):
```

If permission type defaults to "read" here, the below `if not permission_type` would never be true, except when `None` is passed explicitly. Is this the desired behavior?